projector-agent/build.gradle.kts

```
@@ -0,0 +1,145 @@
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact JetBrains, Na Hrebenech II 1718/10, Prague, 14000, Czech Republic
+ * if you need additional information or have any questions.
+ */
+
+import java.util.*
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+
+plugins {
+  kotlin("jvm")
+  `maven-publish`
+}
+
+publishing {
+  publications {
+    create<MavenPublication>("maven") {
+      from(components["java"])
+    }
+  }
+}
+
+kotlin {
+  explicitApi()
+}
+
+val compileKotlin: KotlinCompile by tasks
+val targetJvm: String by project
+compileKotlin.kotlinOptions.jvmTarget = targetJvm
+
+val projectorClientGroup: String by project
+val projectorClientVersion: String by project
+val javassistVersion: String by project
+val agentClassName = "org.jetbrains.projector.agent.MainAgent"
+val launcherClassName = "org.jetbrains.projector.agent.AgentLauncher"
+version = project(":projector-plugin").version
+
+dependencies {
+  implementation("$projectorClientGroup:projector-common:$projectorClientVersion")
+  implementation("$projectorClientGroup:projector-server-core:$projectorClientVersion")
+  implementation("$projectorClientGroup:projector-util-logging:$projectorClientVersion")
+  implementation("$projectorClientGroup:projector-util-loading:$projectorClientVersion")
+  api(project(":projector-awt"))
+  api(project(":projector-server"))
+  implementation("org.javassist:javassist:$javassistVersion")
+}
+
+var serverTargetClasspath: String? = null
+var serverClassToLaunch: String? = null
+var ideaPath: String? = null
+
+rootProject.file("local.properties").let {
+  if (it.canRead()) {
+    ideaPath = Properties().apply { load(it.inputStream()) }.getProperty("projectorLauncher.ideaPath") ?: null
+    serverTargetClasspath = Properties().apply { load(it.inputStream()) }.getProperty("projectorLauncher.targetClassPath") ?: null
+    serverClassToLaunch = Properties().apply { load(it.inputStream()) }.getProperty("projectorLauncher.classToLaunch") ?: null
+  }
+}
+
+tasks.withType<Jar> {
+  manifest {
+    attributes(
+      "Can-Redefine-Classes" to true,
+      "Can-Retransform-Classes" to true,
+      "Agent-Class" to agentClassName,
+      "Main-Class" to launcherClassName,
+    )
+  }
+
+  exclude("META-INF/versions/9/module-info.class")
+  duplicatesStrategy = DuplicatesStrategy.WARN
+
+  from(inline(configurations.runtimeClasspath)) // todo: remove
+}
+
+// Server running tasks
+println("----------- Agent launch config ---------------")
+println("Classpath: $serverTargetClasspath")
+println("ClassToLaunch: $serverClassToLaunch")
+println("------------------------------------------------")
+if (serverTargetClasspath != null && serverClassToLaunch != null) {
+  task("runWithAgent", JavaExec::class) {
```

I see a bit more concise syntax can be used:

```suggestion
  task<JavaExec>("runWithAgent") {
```